test/UnitTest.cs

```diff
@@ -1412,13 +1412,12 @@ public void TestConst()
             sql = "select 1+2*3, 1+2+a1 from a where a1+2+(1*5+1)>2*3 and 1+2=2+1;";
             TU.ExecuteSQL(sql, "7,3;7,4;7,5", out phyplan);
             Assert.AreEqual(0, TU.CountStr(phyplan, "True"));
-            Assert.AreEqual(1, TU.CountStr(phyplan, "7,3+a.a1[0]"));
-            Assert.AreEqual(1, TU.CountStr(phyplan, "a.a1[0]+2+6>6")); // FIXME
+            Assert.AreEqual(1, TU.CountStr(phyplan, "Output: 7,(a.a1[0]+3)"));
+            Assert.AreEqual(1, TU.CountStr(phyplan, "Filter: a.a1[0]>-2"));
```

Comment:
say a.a1+b.b1=2: if we can arrange it to a.a1=2-b.b1 then a hash join is applicable otherwise, only NLJ is possible. So we may need a different than "put const on right side" rule to rewrite for join condition?